src/fsharp/tainted.fs

```diff
@@ -77,19 +77,19 @@ type internal TypeProviderError
             for msg in errors do
                 f (new TypeProviderError(errNum, tpDesignation, m, [msg], typeNameContext, methodNameContext))
 
-type TaintedContext = { TypeProvider: ITypeProvider; TypeProviderAssemblyRef: ILScopeRef; Lock: TypeProviderLock }
+type TaintedContext = { TypeProvider : ITypeProvider; TypeProviderAssemblyRef : ILScopeRef; TypeProviderDesignation: string; Lock : TypeProviderLock }
 
-[<NoEquality>][<NoComparison>] 
-type internal Tainted<'T> (context: TaintedContext, value: 'T) =
```

Comment:
I don't believe we want `Tainted` to be public

Reply:
Tainted is currently used in TaintedEquals methods in provided entities. If we transfer this logic, for example, inside Tainted, then we could make Tainted internal back